### DISCUSSION TITLE
V3 Question

### DISCUSSION THREAD

**Discussion by PaulAtST** [dynamoose/dynamoose#1377] [Q&A]
Hi I am starting up a new project with Dynamoose, and realized some of the features I am looking for are being included in v3. 

I was wondeirng how stable the beta version of Dynamoose v3 currently is or if there is an estimation of when v3 might be conceded stable enough for production?

**Comment by fishcharlie** [comment]
@PaulAtST I believe it's pretty stable. I don't anticipate any more major breaking changes. The biggest issue currently facing v3 is our publishing to npm system. In previous versions everything was contained within one thing and we had a publish script that just worked.

In v3 we separated things out into different packages. And there are cross dependencies with those other packages. We've updated the publish script to work on all those other packages, but it's pretty buggy and error prone. See the links below for more information.

https://github.com/dynamoose/dynamoose/runs/5169048339?check_suite_focus=true
https://github.com/dynamoose/dynamoose/pull/1372

If anyone wants to try to work towards helping solve those things, it'd be greatly appreciated.

I normally work on Dynamoose most during the weekends, and this weekend is super busy for me. So possibly I'll have time to dig into this more next weekend, but I can't make any guarantees on that.

After all of that gets figured out, it's primarily gonna be about testing it in my own projects, and trying to get more people to test the beta. There is a chance we might make a few more undocumented internal properties/methods private, which is already stated in the release notes, and which you shouldn't be using anyway. But other than that, I don't anticipate any more breaking changes in v3. I could be **TOTALLY** wrong here tho. So please take it with a grain of salt.

We really do need more feedback on v3 (positive or negative). I haven't heard enough users experiences with it to know if it's a solid release or not. So I highly encourage more users to test it and report their experiences (even if it's that everything worked well and you love the release). The more constructive feedback here, the better.

**Reply by aukevanleeuwen** [reply]
> If anyone wants to try to work towards helping solve those things, it'd be greatly appreciated.

I've created https://github.com/dynamoose/dynamoose/pull/1384 to fix the problem mentioned. Obviously I can't test the real release process myself though.

**Comment by PaulAtST** [comment]
Unfortunately I got pulled away from DynamoDB for a bit, but I finally got a chance to work on my project a bit more. 

I did notice a perforce issue with v2 and v3 which I added to this issue here: 
[https://github.com/dynamoose/dynamoose/issues/1260](https://github.com/dynamoose/dynamoose/issues/1260)

While getting v3 up and running I did find that the documentation on programmatically configuring dynamo db is incorrect, the user guide still has this: 

```
const ddb = new dynamoose.aws.sdk.DynamoDB({
    "accessKeyId": "AKID",
    "secretAccessKey": "SECRET",
    "region": "us-east-1"
});
```

Which aws.sdk no longer exist in v3. 

I got it to work using this, but not sure if it's the correct way to do it: 
```
const { DynamoDB } = require("@aws-sdk/client-dynamodb")
const dynamooseAws = new DynamoDB({
    accessKeyId: "",
    secretAccessKey: "",
    region: ""
})
dynamoose.aws.ddb.set(dynamooseAws)
```
